Sources/Atomic.swift

```diff
@@ -55,10 +48,12 @@ internal struct UnsafeAtomicState<State: RawRepresentable> where State.RawValue
 	///   - next: The state to transition to.
 	///
 	/// - returns: `true` if the transition succeeds. `false` otherwise.
-	internal func tryTransition(from expected: State, to next: State) -> Bool {
-		return OSAtomicCompareAndSwap32Barrier(expected.rawValue,
-		                                       next.rawValue,
-		                                       value)
+	internal mutating func tryTransition(from expected: State, to next: State) -> Bool {
+		return withUnsafeMutablePointer(to: &value) { valuePointer in
+			return OSAtomicCompareAndSwap32Barrier(expected.rawValue,
+			                                       next.rawValue,
+			                                       valuePointer)
+		}
```

Comment:
I'm not sure this is valid.

This:

> The pointer argument to body is valid only for the lifetime of the closure. Do not escape it from the closure for later use.

Makes me think that the whole construct may not be atomic.